VueJS/index.js

```diff
@@ -8,6 +8,26 @@ new Vue({
       { id: '1', message: 'Foo', },
       { id: '2', message: 'Bar', },
       { id: '3', message: 'Baz', }
-    ]
+    ],
+    currentIndex: 3,       // key for new todo item
+  },
+  methods: {
+    /**
+     * Removes an item from the todo list.
+     * @param {*} key Key of that item.
+     */
+    removeItem(key) {
+      this.todos.splice(this.todos.findIndex(x => x.id === key), 1);
+    },
+    /**
+     * Adds a new item to the list.
+     * The id preincrements from the starting index 3, the message is 'New Item <index>'.
+     */
+    addItem() {
+      this.todos.push({
+        id: (++this.currentIndex).toString(),
```

Comment:
C-style increment of a totally unrelated property during object initialization. I really don't like functions with side effects.

![eww](http://giphygifs.s3.amazonaws.com/media/1Pyi8KuRGJ3AA/giphy.gif)

Try to help your fellow co-workers to read your code. If you want to increment something, make it explicit. Then do the object initialization:

```js
this.currentIndex += 1
// now after your reader has digested this, do your stuff.. 
this.todos.push({
  // ... sometimes it's just good to make things explicit
})
```

But as I said: The whole solution is unnecessary. Utilize VueJS update mechanisms to avoid state:
https://github.com/m-oehme/Systems-Development-and-Frameworks/pull/2#discussion_r339841486